cgp/genome.py

```diff
@@ -346,6 +349,25 @@ def _is_active_input_gene(self, gene_idx: int) -> bool:
         else:
             assert False  # should never be reached
 
+    def _select_gene_indices_for_mutation(self, mutation_rate, len_dna):
```

Comment:
```suggestion
    def _select_gene_indices_for_mutation(self, mutation_rate, len_dna, rng):
```
this function should receive an `np.random.RandomState` instance that should be used instead of `np.random` below. i believe this should also fix the reproducibility issues with multiple processes.